src/types/AST.js

```diff
@@ -2,7 +2,7 @@
 
 type Valtype = 'i32' | 'i64' | 'f32' | 'f64' | 'label';
 type ExportDescr = 'func' | 'table' | 'memory' | 'global';
-type Index = NumberLiteral | Identifier;
+type Index = NumberLiteral | LongLiteral | Identifier;
```

Comment:
The `Index` concept might be to vague here.

For example, the CallInstruction take an index because:
- In WAST `(call $foo)` is an identifier
- In WASM/WATF `(call 0)` is an NumberLiteral (and by spec an unsigned i32)

I will try to clarify it.

Reply:
Refs https://github.com/xtuc/js-webassembly-interpreter/issues/69 where some Identifier will be transformed to NumberLiterals. The idea is to remove `type Index = Identifier` completely in the future.